File: old_backend/important-core-code/javascript/vulnerable-path.js
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical path traversal vulnerabilities for testing Aikido scanner

const express = require('express');
const fs = require('fs');
const path = require('path');
const app = express();

app.get('/download', (req, res) => {
  const filename = req.query.file;
  // VULNERABLE: Direct path traversal
  res.sendFile(filename);
});

app.get('/read', (req, res) => {
  const file = req.query.path;
  // VULNERABLE: No sanitization
  const content = fs.readFileSync('/var/data/' + file, 'utf8');
  res.send(content);
});

app.get('/static', (req, res) => {
  const resource = req.query.resource;
  // VULNERABLE: path.join doesn't prevent traversal
  const filePath = path.join(__dirname, 'public', resource);
  res.sendFile(filePath);
});

app.listen(3000);

File: old_backend/important-core-code/ruby/vulnerable_path.rb
# INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
# This file contains critical path traversal vulnerabilities for testing Aikido scanner

class FilesController < ApplicationController
  def download
    filename = params[:file]
    
    # VULNERABLE: Direct file sending
    send_file "/var/data/#{filename}"
  end
  
  def read
    path = params[:path]
    
    # VULNERABLE: No path validation
    content = File.read(path)
    render plain: content
  end
end
